evdns: add NS,MX,SOA,TXT,SRV,CNAME types of DNS requests

KuzinAndrey opened the pull request:
To create more functional DNS server i need this type of requests, and i decide to add it to the library.

All code was tested with valgrind to prevent any memory leakage.

KuzinAndrey commented:
You can explore project which use this at [dns-blackhole](https://github.com/KuzinAndrey/dns-blackhole)

azat commented on include/event2/dns_compat.h:
```diff
@@ -211,6 +211,108 @@ int evdns_resolve_ipv4(const char *name, int flags, evdns_callback_type callback
 EVENT2_EXPORT_SYMBOL
 int evdns_resolve_ipv6(const char *name, int flags, evdns_callback_type callback, void *ptr);
 
+/**
```
Let's not add compat layer for new functionality

KuzinAndrey replied:
Agree

azat reviewed (commented):
Thank you! Can you add unit tests?

KuzinAndrey committed:
evdns: add NS request type

KuzinAndrey committed:
evdns: add SOA request type

KuzinAndrey committed:
evdns: add MX request type

KuzinAndrey committed:
evdns: add TXT request type

KuzinAndrey committed:
evdns: add SRV request type

KuzinAndrey commented:
@azat i make full rewrite of PR branch:
- One new feature is one commit
- Add unit test to all
I hope you will find time to view my PR and merge it to main branch.

KuzinAndrey committed:
evdns: add CNAME type resolver and DNS_CNAME_CALLBACK_FIRST flag

- Add CNAME type call for resolver.
- Add DNS_CNAME_CALLBACK_FIRST flag for A/AAAA CNAME callback.

In common case A/AAAA callback with CNAME going after main answer, but
user can already process the first callback, free the request handler
and nothing known abount second callback.
To prevent the segmentation fault in CNAME callback we can revert order
of callbacks with this flag and control behaviour by type of request
in callback (A/AAAA or CNAME).

azat commented:
@KuzinAndrey thank you, at first glance looks good, but I need to take a look more carefully. I hope will do it next week. Sorry for the dealy
